x-pack/plugins/inference/common/output/create_output_api.ts

```diff
@@ -9,22 +9,29 @@ import { map } from 'rxjs';
 import { ChatCompleteAPI, ChatCompletionEventType, MessageRole } from '../chat_complete';
 import { withoutTokenCountEvents } from '../chat_complete/without_token_count_events';
 import { OutputAPI, OutputEvent, OutputEventType } from '.';
+import { ensureMultiTurn } from '../ensure_multi_turn';
 
 export function createOutputApi(chatCompleteApi: ChatCompleteAPI): OutputAPI {
-  return (id, { connectorId, input, schema, system }) => {
+  return (id, { connectorId, input, schema, system, messages }) => {
     return chatCompleteApi({
       connectorId,
       system,
-      messages: [
+      messages: ensureMultiTurn([
+        ...(messages || []),
```

Comment:
Do you really think we want that done at that level? We already have that done for `Gemini` at the adapter's level:

https://github.com/elastic/kibana/blob/48e4dcfc9b80e7c8402e6eef6f10adbf7d8c6c44/x-pack/plugins/inference/server/chat_complete/adapters/gemini/gemini_adapter.ts#L153-L164

Which is cleaner, as it can work with the underlying format and append the things the right way instead of injecting `-` messages in the middle (that's not much noise, but still).

If other LLM services have the same limitation / restriction, I would do the patching at the adapter's level for them to, WDYT?

Reply:
yeah you're probably right, I was trying to avoid the overhead of doing it three times, doesn't openai need it as well?

Reply:
IIRC no, openAI doesn't care about it. 

Now regarding the issue about doing it multiple times: I went with the approach to do it at the adapter's level because I felt like doing it during the conversion to the underlying LLM's syntax was more powerful, but if we want to avoid doing it multiple time, we can do that check / transform in the chat API before calling the adapter.